### PR TITLE
Add an spi::config::BitOrder enum

### PR DESCRIPTION
**Pull request by horvbalint**
This PR introduces a new enum spi::config::BitOrder which can be used to specify the order of the bits transfered/received. I tried to test all the variants of the enum, but I am new to embedded development, so I would appreciate if someone else would also test them :)

Thanks @Vollbrecht and @Dominaezzz for the help in the Matrix channel!

### CODE REVIEW

File: src/spi.rs [hunk 1]
```diff
@@ -165,6 +189,7 @@ pub mod config {
         /// See https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/spi_master.html#timing-considerations
         pub write_only: bool,
         pub duplex: Duplex,
+        pub bit_order: BitOrder,
         pub cs_active_high: bool,
         pub input_delay_ns: i32,
     }
```
Comment (Vollbrecht) [inline src/spi.rs:195]: NIT: for consistency maybe add an pub fn bitorder to the Config builder pattern.

**Review by Vollbrecht — APPROVED** [review]
I tested it on an esp32c3 and seems to work as accepted.
MSB First: 
Screenshot from 2023-03-27 22-07-26

LSB First:

Screenshot from 2023-03-27 22-06-09

File: src/spi.rs (hunk 1 above)
Comment (horvbalint) [reply src/spi.rs:195]: Thanks, I missed that 😄 Added it now

**Comment by ivmarkov** [comment]
Thanks!

**Pull request merged**